src/pymmcore_plus/mda/handlers/_ome_zarr_writer.py

```diff
@@ -208,40 +232,50 @@ def frameReady(
         self, frame: np.ndarray, event: useq.MDAEvent, meta: dict | None = None
     ) -> None:
         """Write frame to the zarr array for the appropriate position."""
+        # get the position key to store the array in the group
         key = f'{POS_PREFIX}{event.index.get("p", 0)}'
+
         if key in self._arrays:
             ary = self._arrays[key]
         else:
             # this is the first time we've seen this position
             # create a new Zarr array in the group for it
             if not self._current_sequence:
                 # just in case sequenceStarted wasn't called
-                self._set_sequence(event.sequence)
+                self._current_sequence = event.sequence
 
             if not (seq := self._current_sequence):
                 # This needs to be implemented for cases where we're not executing
-                # an MDASequence.  Or, we need to create a better "mock" MDASequence
+                # an MDASequence. Or, we need to create a better "mock" MDASequence
                 # for generic Iterable[MDAEvent]
                 raise NotImplementedError(
                     "Writing zarr without a MDASequence not yet implemented"
                 )
 
             # create the new array, getting XY chunksize from the frame
-            # and total shape from the sequence
-            axes = (*(k for k in seq.used_axes if k != "p"), "y", "x")
-            shape = (*tuple(v for k, v in seq.sizes.items() if k in axes), *frame.shape)
-            ary = self._new_array(key, shape, frame.dtype, axes)
-
-            # write the MDASequence metadata and xarray _ARRAY_DIMENSIONS to the array
+            # and total shape from the sequence.
+            curr_pos_size = position_sizes(self._current_sequence)[
+                event.index.get("p", 0)
+            ]
+            shape = tuple(curr_pos_size[k] for k in curr_pos_size) + frame.shape
+            used_axes = tuple(curr_pos_size)
+
+            # create the array in the group
+            ary = self._new_array(key, shape, frame.dtype, used_axes)
+
+            # write the MDASequence metadata and xarray _ARRAY_DIMENSIONS to the array.
+            # _ARRAY_DIMENSIONS will be used to get the correct index order when writing
+            # to disk.
             ary.attrs.update(
                 {
                     "useq_MDASequence": json.loads(seq.json(exclude_unset=True)),
-                    "_ARRAY_DIMENSIONS": axes,
+                    "_ARRAY_DIMENSIONS": used_axes,
```

Comment:
this is incorrect.  `_ARRAY_DIMENSIONS` needs to have same length as shape (meaning it needs to include "y" and "x".   This key is part of the xarray zarr spec: https://docs.xarray.dev/en/latest/internals/zarr-encoding-spec.html ... and this change breaks that.

Looks like that wasn't explicitly tested before, but this needs to be changed back to how it used to be (and the tests need to be changed to make sure that `_ARRAY_DIMENSIONS` includes `y` and `x`)... and that might mean you also need to undo the change below that uses `_ARRAY_DIMENSIONS` to get the index